examples/blueprint-construct/index.ts

```diff
@@ -65,14 +66,16 @@ export default class BlueprintConstruct extends cdk.Construct {
             new ssp.addons.VpcCniAddOn(),
             new ssp.addons.CoreDnsAddOn(),
             new ssp.addons.KubeProxyAddOn(),
-            new ssp.addons.OpaGatekeeperAddOn()
+            new ssp.addons.OpaGatekeeperAddOn(),
+            new ssp.addons.KarpenterAddOn(),
```

Comment:
Is not not going to clash with cluster autoscaler which is added on line 57?

Reply:
Probably `console.assert()` in each both addons to ensure the other one is not being used? Also add a unit test to cover this case.

Reply:
Technically you can uses both in same cluster and manage different nodes labels for each

Reply:
Technically they can both be used: if contraints overlap, it's a race to schedule - from my observations Karpenter usually wins out; if node group constraints aren't supported by any Provisioner then CA can schedule unimpeded, although I'm not sure that's a viable use case.

Regardless, I'm of the opinion that we should leave it up to the users to choose between Cluster Autoscaler and Karpenter Add on, in which case assertions should do?

Reply:
I am talking specifically about this configuration when there is a single node group and two autoscalers. I can see how it can work across individual node groups but given that we don't have direct support for multiple node groups it is a moot point. 

The following behavior will suffice now:

If both are registered then provisioning should fail (cdk deploy and synth) with a clear error message. 

Once we add support for multiple node groups, each autoscaler will be modified to optionally accept a nodegroup as a param. Then the conflict detection logic will get a bit more complex.

Reply:
Addressed with errors, unit test successful.